src/store/modules/DownloadModule.ts

```diff
@@ -23,34 +23,39 @@ export const DownloadModule = {
     namespaced: true,
 
     state: (): State => ({
-        allVersions: [],
-        assignId: 0,
-        downloadObject: null, // TODO: Check if the last element of allVersions can be used instead of this
+        allDownloads: [],
     }),
 
     getters: <GetterTree<State, RootState>>{
         activeDownloadCount(state) {
-            const active = state.allVersions.filter(
-                dl => !dl[1].failed && dl[1].progress < 100
+            const active = state.allDownloads.filter(
+                dl => !dl.failed && dl.progress < 100
             );
             return active.length;
         },
+        currentDownload(state) {
+            return state.allDownloads[state.allDownloads.length-1] || null;
+        },
     },
 
     mutations: {
-        setDownloadObject(state: State, downloadObject: DownloadProgress) {
-            state.downloadObject = {...downloadObject};
-        },
-        pushDownloadObjectToAllVersions(state: State, params: { assignId: number, downloadObject: DownloadProgress }) {
-            state.allVersions = [...state.allVersions, [params.assignId, params.downloadObject]];
+        addDownload(state: State, downloadObject: DownloadProgress) {
+            state.allDownloads = [...state.allDownloads, downloadObject];
         },
-        updateDownloadObject(state: State, params: { assignId: number, downloadVersion: [number, DownloadProgress]}) {
-            const newVersions = [...state.allVersions];
-            newVersions[params.assignId] = params.downloadVersion;
-            state.allVersions = newVersions;
+        updateDownload(state: State, downloadObject: DownloadProgress) {
+            const newVersions = [...state.allDownloads];
+            const index = newVersions.findIndex((oldDownloadObject: DownloadProgress) => {
+                return oldDownloadObject.assignId === downloadObject.assignId;
+            });
+            if (index === -1) {
+                throw new R2Error(
```

Comment:
Nitpick: Did you test this? Is the error message shown to user in the error modal like it's written here, or does it get overwritten somewhere along the way? If this error is shown to the user, I think the middle line might be too technical while offering the user no actionable information, nor provides anything for us to debug the issue. Something like "DownloadProgress with id ${oldDownloadObject.assignId} was not found" would at least help in debugging if the value turns out to be undefined or something?